agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/Configuration.java

```diff
@@ -578,22 +596,35 @@ private static boolean isRuntimeAttached() {
   }
 
   public static class SamplingOverride {
-    // TODO (trask) consider making this required when moving out of preview
-    @Nullable public SpanKind spanKind;
+    @Deprecated @Nullable public SpanKind spanKind;
+
+    // TODO (trask) make this required when moving out of preview
+    //   for now the default is both "request" and "dependency" for backwards compatibility
+    @Nullable public TelemetryKind telemetryKind;
+
+    // TODO (trask) should this be named "standalone" (and meaning flipped)
+    //  especially if we aren't going to change meaning of "request" to include unparented INTERNAL
+    //  spans
+    public boolean inRequest = true;
+
     // not using include/exclude, because you can still get exclude with this by adding a second
     // (exclude) override above it
     // (since only the first matching override is used)
     public List<SamplingOverrideAttribute> attributes = new ArrayList<>();
-    public Float percentage;
+    public Double percentage;
     public String id; // optional, used for debugging purposes only
 
+    public boolean isForRequestTelemetry() {
+      return telemetryKind == TelemetryKind.REQUEST
+          || (telemetryKind == null && spanKind != SpanKind.CLIENT);
+    }
+
+    public boolean isForDependencyTelemetry() {
+      return telemetryKind == TelemetryKind.DEPENDENCY
+          || (telemetryKind == null && spanKind != SpanKind.SERVER);
```

Comment:
same here spanKind == SpanKind.Client? what other spanKind is for dependency telemetry?

Reply:
same